events/event_cache.go

```diff
@@ -37,5 +32,14 @@ func (evc *EventCache) Flush() {
 	for _, ei := range evc.events {
 		evc.evsw.FireEvent(ei.event, ei.data)
 	}
-	evc.events = make([]eventInfo, eventsBufferSize)
+	// Clear the buffer by re-slicing its length to zero
+	if cap(evc.events) > len(evc.events)<<1 {
```

Comment:
is `x << 1` supposed to be better for some reason than `x * 2` ?

Reply:
umm I think in my head it said 'order of magnitude', but on reflection its obfuscatory and certainly will compile down to the same thing